Mage.Sets/src/mage/cards/g/GluttonousHellkite.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        List<TargetSacrifice> sacTargets = new ArrayList<>();
+        Player controller = game.getPlayer(source.getControllerId());
+        Object obj = getValue(CastSourceTriggeredAbility.SOURCE_CAST_SPELL_ABILITY);
+        int sacrificeAmount = 0;
+        if (controller == null ) {
+            return false;
+        }
+        if (obj instanceof SpellAbility) {
+            sacrificeAmount = ((SpellAbility) obj).getManaCostsToPay().getX();
+        }
+        if (sacrificeAmount < 1) {
+            return true;
+        }
+        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player != null) {
+                TargetSacrifice target = new TargetSacrifice(sacrificeAmount, StaticFilters.FILTER_PERMANENT_CREATURE);
+                if (target.canChoose(player.getId(), source, game)) {
+                    player.choose(Outcome.Sacrifice, target, source, game);
+                    sacTargets.add(target);
+                }
+            }
+        }
+        for (TargetSacrifice sacTarget : sacTargets) {
+            for (UUID targetId : sacTarget.getTargets()) {
+                Permanent permanent = game.getPermanent(targetId);
+                if (permanent != null) {
+                    permanent.sacrifice(source, game);
+                }
+            }
+        }
+
+        int sacrificeCount = sacTargets.stream().map(TargetImpl::getTargets).map(List::size).reduce(0, Integer::sum);
+        for (Effect effect : source.getEffects()) {
+            if (effect instanceof GluttonousHellkiteReplacementEffect) {
+                effect.setValue("GluttonousHellkiteCounters", sacrificeCount);
+            }
+        }
+        return true;
+    }
+}
+
+class GluttonousHellkiteReplacementEffect extends ReplacementEffectImpl {
+
+    GluttonousHellkiteReplacementEffect() {
+        super(Duration.OneUse, Outcome.BoostCreature);
+        staticText = "{this} enters the battlefield with two +1/+1 counters on it for each creature sacrificed this way";
+    }
+
+    private GluttonousHellkiteReplacementEffect(final GluttonousHellkiteReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        return event.getTargetId().equals(source.getSourceId());
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Object object = this.getValue("GluttonousHellkiteCounters");
+
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        if (permanent == null) {
+            permanent = game.getPermanentEntering(source.getSourceId());
```

Comment:
won't this always be an entering permanent?

Reply:
I'm not sure, I copied this code from another card

Reply:
if you copy code like that, in future please add a comment referencing the source - helps with review

Reply:
Anyway, since this is a replacement effect, when it replaces the `EventType.ENTERS_THE_BATTLEFIELD`, `getPermanent()` will always be null, and `getEnteringPermanent()` is the relevant thing.

But there's an even better technique which is `((EntersTheBattlefieldEvent) event).getTarget()`

Reply:
I can start doing that

Reply:
I have swapped to `((EntersTheBattlefieldEvent) event).getTarget()` but now it doesn't even appear as a creature.
With the replacement effect do I have to add the `game.getBattlefield().addPermanent(permanent);` logic as well?

Reply:
ugh, well, if it doesn't work, then just leave it how you had it I guess. sorry.

Reply:
Weird I changed nothing rebuild and it's working now